### DISCUSSION TITLE
Central chat

### DISCUSSION THREAD

**Discussion by SZEINER** [Leantime/leantime#1382] [Plugins]
Work on centralized chat inspired by Teamwork chat app from image.

![obrazek](https://user-images.githubusercontent.com/77299927/216816387-05677377-1c67-45d5-b5c0-17e5de5641a7.png)

**Work:**

- chat with members

**Future plan:**

- groups ( create, leave, delete)
- videochat
- Attach file from central drive
- Will be added...

**Comment by marcelfolaron** [comment]
Thanks, that would be cool. Before you get started let's sync up on Discord. I want to make sure we align on architecture (polling vs sse vs sockets). I would also suggest to start very simple and do more of a linkedIn style chat. 

![image](https://user-images.githubusercontent.com/47038998/216837542-3a4bb21d-c817-4c37-b61a-76081b24b270.png)

**Reply by marcelfolaron** [reply]
With small chatboxes on the bottom and user picker right next to it.

**Reply by SZEINER** [reply]
Yeah linkedin style is very interesting. We look at this point. Its very simple make plugins under this system.

**Comment by bluebrad** [comment]
or maybe API options to link to discord or google chat. there is also https://zulip.com/ | https://github.com/chat-sdk | http://www.chat21.org/ that maybe can be integrated. alternatively have a private "$@" -  option to comment on a task or any ware in the system using the combo of that tag will allow only the tagged people to see that comment.

**Comment by marcelfolaron** [comment]
We have slack, discord, mattermost and zulip integrations right now.
Updates will be pushed to those channels using webhooks.
Is that what you are looking for?

**Reply by bluebrad** [reply]
Not exactly i have tried that tool but i feel there should be a simple embedded chat that we can "@" tag personnel and "#" tag tasks or milestones or bring in discord and slack deeper. to where its actually embedded.

**Reply by bluebrad** [reply]
Maybe something on the dashboard that will allow for a running chat a project box that is closer to a basic chat box. and when "@" tag you will have a chat notification at the top menu.
